core/src/test/scala/org/genivi/sota/core/CampaignResourceSpec.scala

```diff
@@ -323,4 +366,52 @@ class CampaignResourceSpec extends FunSuite
 
     getCampaignsOk().filter(_.id == id).map(_.createdAt) shouldEqual createdAt
   }
+
+  test("campaign with static delta changes to 'in preparation' after launch") {
+    createCampaignWithStaticDeltaOk().meta.status shouldBe CampaignStatus.InPreparation
+  }
+
+  test("campaign with static delta cannot be edited after launch") {
+    val campaign = createCampaignWithStaticDeltaOk()
+
+    setGroups(campaign.meta.id, createRandomGroups(), StatusCodes.Locked)
+
+    val fromCommit = "162ede9bdc7b53f7497e98af4381f95fde24667fc829aea8d933b70afedb7a0a"
+    setDeltaFrom(campaign.meta.id, Some(createRandomTreehubPackage(fromCommit)), StatusCodes.Locked)
+  }
+
+  test("campaign with static delta changes state to active after delta generation succeeds") {
+    val campaign = createCampaignWithStaticDeltaOk()
+
+    async {
+      val fromCommit: Commit = Refined.unsafeApply("c62ede9bdc7b53f7497e98af4381f95fde24667fc829aea8d933b70afedb7a0a")
+      val toCommit: Commit = Refined.unsafeApply("c62ede9bdc7b53f7497e98af4381f95fde24667fc829aea8d933b70afedb7a0b")
+      val size = 100
+      await(new DeltaListener(deviceRegistry, updateService, messageBus)
+        .generatedDeltaAction(GeneratedDelta(campaign.meta.id.underlying, Namespaces.defaultNs, fromCommit, toCommit,
+                                             OstreeSummary(Array.emptyByteArray), Uri(), size)))
+      val camp = fetchCampaignOk(campaign.meta.id)
+      camp.meta.status shouldBe CampaignStatus.Active
+      camp.meta.size shouldBe size
```

Comment:
Turns out you need to have `.futureValue` at the end of the async block. I rewrote them to not use `async`, to avoid bugs like this in future, otherwise your test always succeeds. Great catch though, it turns out there were multiple bugs I hadn't noticed once I fixed the test.